### PR TITLE
Date picker sample

### PR DESCRIPTION
**Pull request by kul3r4**
Added some basic samples. I think we could start from this sample and then add more use cases later.

### CODE REVIEW

File: src/20_Components/amp-date-picker.html [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    <input amp-date-placeholder-end placeholder="End date" id="end" />
+    <template type="amp-mustache" info-template>
+      <span [text]="'You picked ' + startDate + ' as start date and ' + endDate + ' as end date.'">You will see your chosen dates here.</span>
+    </template>
+  </amp-date-picker>
+
+
+  </div>
+
+  <!-- #### External configuration -->
```
Comment (sebastianbenz) [inline src/20_Components/amp-date-picker.html]: use title case for headings

File: src/20_Components/amp-date-picker.html [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+      "templates": [{
+      "id": "low",
+      "dates": [
+        "2017-11-01", "2017-11-03", "2017-11-05", "2017-11-06",
+        "FREQ=WEEKLY;DTSTART=20171101T150000Z;COUNT=30;WKST=MO;BYDAY=TU,SA"
+        ]
+      }]
+    }
+```
+  and we set the `src` of amp-date-picker with the filename amp-date-picker.json -->
```
Comment (sebastianbenz) [inline src/20_Components/amp-date-picker.html]: * Can you mark amp-date-picker and amp-date-picker.json with "`"
* s/and/Then

File: templates/css/example.css [hunk 3]
```diff
@@ -96,7 +96,6 @@ code {
 }
 .abe-preview {
   background: #fff;
-  overflow: hidden;
```
Comment (sebastianbenz) [inline templates/css/example.css]: why did you remove this?

File: src/20_Components/amp-date-picker.html [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    right: 0;
+    width: 0;
+    height: 0;
+    border: 10px solid transparent;
+    border-top-color: red;
+    border-bottom: 0;
+    border-right: 0;
+  }
+
+  .DateRangePicker, .SingleDatePickerInput {
```
Comment (sebastianbenz) [inline src/20_Components/amp-date-picker.html]: What are these rules used for?

File: src/20_Components/amp-date-picker.html [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+      class="mr1 ml1">
+      <input amp-date-placeholder placeholder="Pick a date!!"/>
+      <template type="amp-mustache" info-template>
+        <span [text]="date != null ? 'You picked ' + date + '.' : 'You will see your chosen date here.'">You will see your chosen date here.</span>
+      </template>
+    </amp-date-picker>
+  </div>
+  <!-- #### Date Range Picker -->
+  <!--
+    `amp-date-picker` with `type=range` can be used for selecting a date range. The default css will disable
```
Comment (sebastianbenz) [inline src/20_Components/amp-date-picker.html]: Does the css really disables this? Suggestion: "By default, amp-date-picker disables past dates, use ..."

File: src/20_Components/amp-date-picker.html [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+      locale="en"
+      format="Y-MM-DD"
+      number-of-months="1"
+      show-clear-date
+      keep-open-on-date-select
+      id="simple-date-picker"
+      class="mr1 ml1">
+      <input amp-date-placeholder placeholder="Pick a date!!"/>
+      <template type="amp-mustache" info-template>
+        <span [text]="date != null ? 'You picked ' + date + '.' : 'You will see your chosen date here.'">You will see your chosen date here.</span>
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: date is a variable set by `amp-bind`. Please take a look at 
     ` on="select:AMP.setState({date: event.date, dateType: event.id})"`

File: src/20_Components/amp-date-picker.html (hunk 5 above)
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: Yes it does.

File: templates/css/example.css (hunk 3 above)
Comment (kul3r4) [reply templates/css/example.css]: Because otherwise the calendar was not showing as an overlay.

File: src/20_Components/amp-date-picker.html (hunk 2 above)
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: done

File: src/20_Components/amp-date-picker.html (hunk 1 above)
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: done

File: src/20_Components/amp-date-picker.html [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+      <template type="amp-mustache" info-template>
+        <span
+          [text]="'You picked ' + date + '.' +
+          (dateType == 'tacos1' ? ' Happy Taco Tuesday!' : '')"
+        >You will see your chosen date here.</span>
+      </template>
+    </amp-date-picker>
+  </div>
+  <!--
+    `amp-date-picker` supports a range of attributes, for example `highlighted` and `blocked`. Find the complete list int the official [doc](https://github.com/ampproject/amphtml/tree/master/extensions/amp-date-picker/0.1).
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: done

File: src/20_Components/amp-date-picker.html [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+      class="mr1 ml1">
+      <input amp-date-placeholder placeholder="Pick a date!!"/>
+      <template type="amp-mustache" info-template>
+        <span [text]="date != null ? 'You picked ' + date + '.' : 'You will see your chosen date here.'">You will see your chosen date here.</span>
+      </template>
+    </amp-date-picker>
+  </div>
+  <!--
+    In addition, you can also use `amp-mustache` to create templates for custom date markup.
+    Here we are using a taco icon in place of the number of the month with a frequency of 2 weeks every Tuesday.
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: done

File: src/20_Components/amp-date-picker.html [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+      id="simple-date-picker"
+      class="mr1 ml1">
+      <input amp-date-placeholder placeholder="Pick a date!!"/>
+      <template type="amp-mustache" info-template>
+        <span [text]="date != null ? 'You picked ' + date + '.' : 'You will see your chosen date here.'">You will see your chosen date here.</span>
+      </template>
+    </amp-date-picker>
+  </div>
+  <!--
+    In addition, you can also use `amp-mustache` to create templates for custom date markup.
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: done

File: src/20_Components/amp-date-picker.html [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+<!doctype html>
+<html ⚡>
+<head>
+  <meta charset="utf-8">
+  <!-- #### Introduction
+    AMP date picker...add intro here
+  -->
+  <!-- #### Setup -->
+  <!--
+    Include the amp-date-picker extended component.
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: done

File: src/20_Components/amp-date-picker.html [hunk 11]
```diff
@@ -0,0 +1,202 @@
+<!---{
+    "experiments": ["amp-date-picker"]
+}--->
+<!doctype html>
+<html ⚡>
+<head>
+  <meta charset="utf-8">
+  <!-- #### Introduction
+    AMP date picker...add intro here
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: done

File: templates/css/example.css (hunk 3 above)
Comment (sebastianbenz) [reply templates/css/example.css]: Does this affect the way previews are rendered in other samples?

File: src/20_Components/amp-date-picker.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply src/20_Components/amp-date-picker.html]: So mustache cannot be used together with the date picker? I find that very surprising (and confusing).

File: src/20_Components/amp-date-picker.html (hunk 4 above)
Comment (sebastianbenz) [reply src/20_Components/amp-date-picker.html]: Can you document these classes and what you can use them for.

Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: I added a line explaining those are inherited from react

File: src/20_Components/amp-date-picker.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: Yes it can. Look at line 220.

File: templates/css/example.css (hunk 3 above)
Comment (kul3r4) [reply templates/css/example.css]: I haven't found any problem yet

File: src/20_Components/amp-date-picker.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (kul3r4) [reply src/20_Components/amp-date-picker.html]: Also, I see where is your confusion!
there are 2 types of templates:
info-template and date-template.
Inside date-template you can access mustache variables, like {{d}} for deciding how to style a day in the calendar.
Inside info-template you can access amp-bind variables, this is for showing an info panel just below the calendar.

@cvializ correct me if I am wrong here!
